### DISCUSSION TITLE
Roadmap?

### DISCUSSION THREAD

**Discussion by magick93** [hofstadter-io/hof#170] [General]
This is a really useful and powerful project. And I hope it continues to get the attention it deserves.

I'm wondering if there is a roadmap of features, or what is the plan to reaching v1.0.0?

**Comment by verdverm** [comment]
Thanks @magick93 

There is not a hard list of features on the way to v1. I think the main potential blocker is wanting to have a CUE v1 for a HOF v1. I think this should be the plan. I'm basing this on Terraform that had wide, production usage prior to v1.

The main features I'd like to see for v0.7 are:

- hof/datamodel revamp (WIP)
- hof/flow revamp (TBD)
- expose structural commands (pick, mask, diff, etc...)
- maybe embed `cue` commands like def, eval, export...
- examples and documentation (WIP)

At the v0.7 point, I think that the main features are pretty well determined. There are a lot of little things that depend on CUE improvements. I think most of the other things will work themselves out by then. 


Are there any main features that are missing?

---

2 big items to be addressed

1. Renaming or moving things
2. Bringing existing code into hof code gen

**Reply by magick93** [reply]
I'd like to see the docs improved. There are several TODOs and holes to plug there. 

I'm happy to help with these.

**Reply by verdverm** [reply]
That would be great. There is a lot of movement on the documentation. We are working on the contributor docs this week too!

btw, there is a https://next.hofstadter.io ... planning to publish the latest later today, once the open PR for code-gen gets in
